CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+helps, and credit will always be given.
+
+You can contribute in many ways:
+
+## Types of Contributions
+
+### Report Bugs
+
+Report bugs at https://github.com/GNS-Science/nzshm-common-py/issues.
+
+If you are reporting a bug, please include:
+
+- Your operating system name and version.
+- Any details about your local setup that might be helpful in troubleshooting.
+- Detailed steps to reproduce the bug.
+
+### Fix Bugs
+
+Look through the GitHub issues for bugs. Anything tagged with "bug" and "help
+wanted" is open to whoever wants to implement it.
+
+### Implement Features
+
+Look through the GitHub issues for features. Anything tagged with "enhancement"
+and "help wanted" is open to whoever wants to implement it.
+
+### Write Documentation
+
+nzhsm-common-py could always use more documentation, whether as part of the
+official nzhsm-common-py docs, in docstrings, or even on the web in blog posts,
+articles, and such.
+
+### Submit Feedback
+
+The best way to send feedback is to file an issue at https://github.com/GNS-Science/nzhsm-common-py/issues.
+
+If you are proposing a feature:
+
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Remember that this is a volunteer-driven project, and that contributions
+  are welcome :)
+
+## Get Started!
+
+Ready to contribute? Here's how to set up `nzhsm-common-py` for local development.
+
+1. Fork the `nzhsm-common-py` repo on GitHub.
+2. Clone your fork locally
+
+   ```
+   $ git clone git@github.com:your_name_here/nzhsm-common-py.git
+   ```
+
+3. Ensure [poetry](https://python-poetry.org/docs/) is installed.
+4. Install dependencies and start your virtualenv:
+
+   ```
+   $ poetry install -E test -E doc -E dev
+   ```
+
+5. Create a branch for local development:
+
+   ```
+   $ git checkout -b name-of-your-bugfix-or-feature
+   ```
+
+   Now you can make your changes locally.
+
+6. When you're done making changes, check that your changes pass the
+   tests, including testing other Python versions, with tox:
+
+   ```
+   $ poetry run tox
+   ```
+
+7. Commit your changes and push your branch to GitHub:
+
+   ```
+   $ git add .
+   $ git commit -m "Your detailed description of your changes."
+   $ git push origin name-of-your-bugfix-or-feature
+   ```
+
+8. Submit a pull request through the GitHub website.
+
+## Pull Request Guidelines
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+2. If the pull request adds functionality, the docs should be updated. Put
+   your new functionality into a function with a docstring, and add the
+   feature to the list in README.md.
+3. The pull request should work for Python 3.6, 3.7, 3.8 and 3.9. Check
+   https://github.com/GNS-Science/nzhsm-common-py/actions
+   and make sure that the tests pass for all supported Python versions.
+
+## Tips
+
+```
+$ poetry run pytest tests/????.py
+```
+
+To run a subset of tests.
+
```

Comment:
Maybe also add the following:

<hr>

```shell
$ poetry run pytest -k keyword
```

To run any test functions with "keyword" in their names.

<hr>

I find that incredibly useful when I'm writing individual tests and don't want too much extra noise on the console.

Reply:
OK  - this page is really just boilerplate ATM .. we don't want it to become a poetry tutorial :)